base/applications/taskmgr/perfpage.c

```diff
@@ -168,30 +169,19 @@ PerformancePageWndProc(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
         hPerformancePageMemUsageHistoryGraph = GetDlgItem(hDlg, IDC_MEM_USAGE_HISTORY_GRAPH);
         hPerformancePageCpuUsageHistoryGraph = GetDlgItem(hDlg, IDC_CPU_USAGE_HISTORY_GRAPH);
 
-        GetClientRect(hPerformancePageCpuUsageHistoryGraph, &rc);
-        /*  create the control */
-        /* PerformancePageCpuUsageHistoryGraph.Create(0, rc, hDlg, IDC_CPU_USAGE_HISTORY_GRAPH); */
-        GraphCtrl_Create(&PerformancePageCpuUsageHistoryGraph, hPerformancePageCpuUsageHistoryGraph, hDlg, IDC_CPU_USAGE_HISTORY_GRAPH);
-        /*  customize the control */
-        GraphCtrl_SetRange(&PerformancePageCpuUsageHistoryGraph, 0.0, 100.0, 10);
-/*         PerformancePageCpuUsageHistoryGraph.SetYUnits("Current") ; */
-/*         PerformancePageCpuUsageHistoryGraph.SetXUnits("Samples (Windows Timer: 100 msec)") ; */
-/*         PerformancePageCpuUsageHistoryGraph.SetBackgroundColor(RGB(0, 0, 64)) ; */
-/*         PerformancePageCpuUsageHistoryGraph.SetGridColor(RGB(192, 192, 255)) ; */
-/*         PerformancePageCpuUsageHistoryGraph.SetPlotColor(RGB(255, 255, 255)) ; */
-        GraphCtrl_SetBackgroundColor(&PerformancePageCpuUsageHistoryGraph, RGB(0, 0, 0)) ;
-        GraphCtrl_SetGridColor(&PerformancePageCpuUsageHistoryGraph, RGB(0, 128, 64));
-
-        GraphCtrl_SetPlotColor(&PerformancePageCpuUsageHistoryGraph, 0, RGB(0, 255, 0)) ;
-        GraphCtrl_SetPlotColor(&PerformancePageCpuUsageHistoryGraph, 1, RGB(255, 0, 0)) ;
-
-
-        GetClientRect(hPerformancePageMemUsageHistoryGraph, &rc);
-        GraphCtrl_Create(&PerformancePageMemUsageHistoryGraph, hPerformancePageMemUsageHistoryGraph, hDlg, IDC_MEM_USAGE_HISTORY_GRAPH);
-        GraphCtrl_SetRange(&PerformancePageMemUsageHistoryGraph, 0.0, 100.0, 10) ;
-        GraphCtrl_SetBackgroundColor(&PerformancePageMemUsageHistoryGraph, RGB(0, 0, 0)) ;
-        GraphCtrl_SetGridColor(&PerformancePageMemUsageHistoryGraph, RGB(0, 128, 64)) ;
-        GraphCtrl_SetPlotColor(&PerformancePageMemUsageHistoryGraph, 0, RGB(255, 255, 0)) ;
+        /*  create the controls */
+	fmt.clrBack = RGB(0, 0, 0);
+	fmt.clrGrid = RGB(0, 128, 64);
+	fmt.clrPlot0 = RGB(0, 255, 0);
+	fmt.clrPlot1 = RGB(255, 0, 0);
+	fmt.GridCellWidth = fmt.GridCellHeight = 12;
+	fmt.DrawSecondaryPlot = TaskManagerSettings.ShowKernelTimes;
+	TmgcCreate(&PerformancePageCpuUsageHistoryGraph, hPerformancePageCpuUsageHistoryGraph, hDlg, &fmt);
+	fmt.clrPlot0 = RGB(255, 255, 0);
+	fmt.clrPlot1 = RGB(168, 128, 255);
+	fmt.DrawSecondaryPlot = TRUE;
+	TmgcCreate(&PerformancePageMemUsageHistoryGraph, hPerformancePageMemUsageHistoryGraph, hDlg, &fmt);
```

Comment:
```suggestion
	if (!TmgcCreate(&PerformancePageMemUsageHistoryGraph, hPerformancePageMemUsageHistoryGraph, hDlg, &fmt))
	{
	    EndDialog(hDlg, 0);
	    return FALSE;
	}
```

Reply:
and how this overcomplication is better, than `PostQuitMessage`? if there is no resources for allocating 1000 bytes, then, it's better to just exit.

Reply:
I tend to agree with @HBelusca regarding these suggestions. The Task Manager is supposed to be able to end processes which ate a lot of memory. Fail-fast principle is not really acceptable here.

Reply:
omitted this. so, you want to change `PostQuiteMessage()` to `EndDialog()` as hermes suggested? it's easy, I am outside of computer, so either I change that, when am back or you can do it by yourself. but please, no need to change `TmgcCreate()` prototype, you can call `EndDialog()` directly from it, without returning TRUE/FALSE, because `hDlg` HWND is passed to it as `hParentWnd`.

imo, failing when there is no memory for 1024 bytes allocation is not "falling fast", in such disastrous cases taskmgr would be probably unusably hanging for 100s other reasons and killing offendind procrsses would be only oossible from `cmd.exe`, but then, this argunent is another vanity. let it be `EndDialog()` if you wish. I agree on this in exchange of switching the secondary plot for menory graph to purple back. :D kidding.

Reply:
> but please, no need to change TmgcCreate() prototype, you can call EndDialog() directly from it, without returning TRUE/FALSE, because hDlg HWND is passed to it as hParentWnd.

Which shows a badly written code. Your graph helper functions expect to be inside a dialog. What if I wanted to reuse the graph code (or librarize it somewhere) to be able to plot graphs on a Win32 app that doesn't use dialogs? EndDialog() would be of no effect, and I would have to modify that code. Instead, don't make your helper functions suppose how the user of these is using them.

Reply:
Yeah despite I understand your motive to make it more optimized, Hermes has the point of better abstraction for the graph plotting code.

I believe we're following this idea in from the start (not only in Task Manager), that's why plotter functions are following common controls naming scheme. I guess I'm about to restore the naming, while hoping you wouldn't be offended by it. :D

After all the purpose of this PR is to preserve graphs history on resizes.